docs/data/toolpad/getting-started/why-toolpad.md

```diff
@@ -0,0 +1,81 @@
+# Why Toolpad?
+
+<p class="description">Learn how Toolpad approaches building internal tools, and how it can be useful to you.</p>
+
+## TL;DR
+
+- **Higher-level API**. Toolpad operates at a higher abstraction level than Material UI. It sets healthy UI constraints that prevent common pitfalls and anti-patterns.
+- **Drag-and-drop builder**. Toolpad comes with a drag-and-drop builder to simplify UI building. The Drag-and-drop state is persisted in a yaml file so you can fully control the visual output with code.
+- **Less boilerplate** so you can focus on the essential parts of the app.
+- **Run alongside existing code**. It integrates well with your existing code. You can use your database models, client libraries, secrets, and bespoke components directly.
+- **Single source of truth**. Because Toolpad runs alongside your existing code, it lets you maintain a single source of truth for all business logic.
+- **Integrates well with your development lifecycle**. Your project lives on your file system. You can use your own IDE, version control system, deployment target, CLI tools.
+
+On reviewing multiple products and approaches, a few recurring areas of concern were noticed. Let's examine them in detail:
+
+## Trade-offs with pro-code tools
+
+### Compromised delivery speed
+
+Every organization aspires to achieve rapid delivery, but to accommodate potential use cases, developers are often compelled to opt for flexible pro-code solutions, which typically come at the expense of speed. These tools offer lower abstraction levels for more control, but handling and scaling them soon becomes an unavoidable maintenance overhead.
+
+### More lines of code
+
+Lots of boilerplate HTML, CSS, and JavaScript glue code is needed just to get backend data to the browser and hooked up to data displaying components.
+
+### Lower-level abstraction
+
+Building internal tools requires knowledge of front-end framework. While it may interest some, most internal tool developers find the lower-level APIs, HTML and CSS an extra burden to maintain.
+
+## Trade-offs with low-code tools
+
+There exist many low-code code tools specifically for internal tools. They are proven to be productive but come with some trade-offs, as mentioned below:
+
+### Codebase fragmentation and duplication
+
+Frequently, developers invest valuable time in hunting down particular versions of business logic and then maintaining them in multiple places. When working with external low-code/no-code tools, code/logic duplication can often arise as a consequence of design decisions. Not having a unified reference point within a codebase compromises its scalability as well as ease of maintenance and debugging.
+
+### Extensibility
+
+Low-code tools are hard to extend. If there's something the tool can't do out of the box, you can't bring your own code with you and integrate it seamlessly.
+
```

Comment:
```suggestion
### Duplication and fragmentation

When working with external low-code or no-code tools, the duplication of code and logic can frequently occur because of a lack of extensibility. Developers invest valuable time in hunting down specific versions of business logic and maintaining them in multiple places. This lack of a unified reference point within the codebase can compromise its scalability, making maintenance and debugging more challenging.
```